torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -555,11 +555,23 @@ def reorder_and_filter(inputs, layout_or_out):
                 assert len(input_indices) >= 2
                 return [inputs[idx] for idx in input_indices], layout_or_out
 
+        new_inputs, new_layout = reorder_and_filter(input_nodes, layout)
+        assert isinstance(new_inputs[1].layout, ir.FixedLayout)
+        # Note that MKLDNN Tensor gets the wrong stride, thus
+        # don't use it for MKLDNN Tensor
+        view_size = new_inputs[1].layout.size
+        view_stride = new_inputs[1].layout.stride
+        view_offset = new_inputs[1].layout.offset
+
         def maybe_to_dense(inputs, layout_or_out):
             new_inputs = list(inputs)
             if isinstance(inputs[1], torch.Tensor):
                 W = inputs[1]
-                new_inputs[1] = W.to_dense() if W.is_mkldnn else W
+                if W.is_mkldnn:
+                    new_inputs[1] = W.to_dense()
+                else:
+                    # Assume W has been unwrap view
+                    new_inputs[1] = W.as_strided(view_size, view_stride, view_offset)
```

Comment:
Does it make more sense to move this shape manipulation code into `normalize_shape` function?

Reply:
Thanks for the comment and move this into `normalize_shape`. Please help to take a look again.